splitgpg2/__init__.py

```diff
@@ -322,6 +322,24 @@ def load_config(self, config) -> None:
         for option in config:
             if option not in supported_options:
                 self.log.warning('Unsupported config option: %s', option)
+        self.log.info('Using GnuPG home directory %s', self.gnupghome)
+        try:
+            subprocess.check_call(('mkdir', '-pm0700', '--', self.gnupghome))
+        except subprocess.CalledProcessError:
+            self.log.error('GnuPG home directory %s cannot be created!', self.gnupghome)
+            raise ValueError()
+        if 'isolated_gnupghome_dirs' not in config:
```

Comment:
IMO it should be possible to use this feature even with `isolated_gnupghome_dirs`. It means:
 - there should be a separate option for disabling this feature (as discussed: this "subkeys only keyring" should be enabled by default)
 - when enabled, it should use some other dir for the keyring (subdir of the full keyring would be convenient, but I'm not sure if allowed - will gpg be confused by extra subdir in its homedir?)

Furthermore, there should be a check if updating subkeys-only keyring is needed (based on mtime likely). And if so - it should create it from scratch - so it would also remove keys no longer present in the original keyring (which is yet another reason to use separate keyring for this feature - to not accidentally remove user's primary keyring).